### ISSUE TITLE
Add regularization

### ISSUE THREAD

**Issue by mnwright** [bips-hb/neuralnet#11]
We should add some kind of regularization such as L2 regularization.

**Comment by alexsiormpas** [comment]
Thanks Marvin, this improvement will help a lot!

**Comment by alexsiormpas** [comment]
In my project I have 1 hidden layer. I've constructed a solution that loops through various architectures and finds the optimal number of nodes that minimizes the cross-validation error. Would that be considered a form of regularization similar to drop-out?

**Comment by mnwright** [comment]
No, I don't think so. Btw., I think we should have both dropout and L1/L2 regularization in the package.

**Comment by cgeorges22** [comment]
Is this still something that is in the pipeline? Thanks.

**Comment by mnwright** [comment]
I still think it would be nice to have it but we currently do not have the resources to further develop this package. A PR would be very much welcome!
